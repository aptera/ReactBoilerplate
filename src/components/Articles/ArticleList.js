import React, {Component} from 'react';
import Article from './Article';
import values from 'object.values';



class ArticleList extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            articles: props.articles,
            authors: props.authors
        };
    }
    
    articleActions = {
        lookupAuthor: (authorId) => this.state.authors[authorId],
    }

    render(){
        return (
            <div style={{ textAlign: 'center', margin: '20px' }}>
                <div className="articles">
                    {values(this.state.articles).map((article) => 
                        <Article
                            key={article.id}
                            article={article}
                            actions={this.articleActions}
                        />
                    )}
                </div>
            </div>
        );
    } 
}

export default ArticleList;