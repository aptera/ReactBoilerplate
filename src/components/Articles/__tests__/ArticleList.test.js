import React from 'react';
import ArticleList from '../ArticleList';

jest.mock('../Article', () => {
    const article = () => <div />;
    return article;
});

import renderer from 'react-test-renderer';

describe('The ArticleList', () => {

    let props;

    beforeEach(() => {
        props = {
            articles: [
                {
                    'id': '1',
                    'title': 'Article One',
                    'date': '01/01/2018',
                    'authorId': '1',
                    'body': 'This is an article about Ones\'s'
                },
                {
                    'id': '2',
                    'title': 'Article Two',
                    'date': '01/05/2018',
                    'authorId': '1',
                    'body': 'This is an article about Two\'s'
                },
                {
                    'id': '3',
                    'title': 'Article Three',
                    'date': '02/06/2018',
                    'authorId': '1',
                    'body': 'This is an article about Three\'s'
                }
            ]
        };
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <ArticleList
                {...props}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});