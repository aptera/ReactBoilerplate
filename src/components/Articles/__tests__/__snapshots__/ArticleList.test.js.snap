// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ArticleList renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "20px",
      "textAlign": "center",
    }
  }
>
  <div
    className="articles"
  />
</div>
`;
